docs/dev/developer-guides/bridging/l1-l2-interop.md

```diff
@@ -1,72 +1,52 @@
+## Priority queue
 
-### Sending messages
+zkSync Era supports general smart contract computation which includes complex operations. For this reason, some principles had to evolve in order to preserve the stability of the network.
 
-Each message sent from L2 to L1 contains the sender's address and the message itself. The length of the message can be arbitrarily large, but the longer the message, the more expensive it will be to send. The operator must include all messages for the corresponding merkle root (see next paragraph). Hence, all the messages are publicly available, and one does not have to rely on the operator to reveal them.
+1. All transactions types are now supported by the priority queue. 
 
-### Reading messages
+    In Lite, users may have had their funds locked on an L2 smart contract, and not on their own L2 account. Now, before moving funds to L1, they can execute a transaction on the zkSync Era network which releases the funds from the smart contract so they can move them.
 
-Every message sent can be read on-chain. Moreover, it is possible to prove that a message has been sent in a specific L2 block. To make such proof as cheap as possible for both the user and the operator, we store all messages, for each L2 block, in a merkle tree. Accordingly, any L1 smart contract can consume the message sent by providing proof of inclusion in some L2 block. Proof can be generated based only on the data that the operator sent to the zkSync L1 smart contract. The proof can also be obtained via [the API](../../../api/api.md#zks-getl2tol1msgproof).
+    :::info
+    The withdraw functionality in zkSync Era replaces `FullExit` in Lite.
+    :::
 
-### Summary on L2->L1 messaging
+2. The priority queue must be fully permissionless to prevent malicious activity. For example, malicious users might send multiple transactions which push up the block gas limit to unworkable levels. To mitigate against this, submitting transactions to the priority queue is no longer free and users must pay a fee to the operator. 
 
-- L2 -> L1 communication requires one transaction on L2 and one on L1.
-- Messages can be of arbitrary length.
-- All the data needed for proving message inclusion in an L2 block can always be restored from Ethereum. However, the easiest way is to request proof from the operator via API.
+    The fee for a transaction is equal to `txBaseCost * gasPrice`. The `gasPrice` is the gas price of the user's transaction, while `txBaseCost` is the base cost for the transaction and depends on the transaction parameters (e.g. `gasLimit` for an `Execute` transaction).
```

Comment:
The fee is equal to `txBaseCost`

Reply:
this whole paragraph is now dubious and needs rewriting (by the way, this has been live for at least [7 months](https://github.com/matter-labs/zksync-web-era-docs/blob/ad8d4de9d9415cbe5db4befb0bcd7487875ed373/docs/dev/developer-guides/bridging/l1-l2-interop.md))

Reply:
Also, here @barakshani ?

Reply:
I'm not sure what exactly txBaseCost stands for (it's does not exist on L1), Stas should know better.

Reply:
@kailash-manasarovar we call the `baseCost` the minimal amount of ETH that needs to be paid for the L1->L2 transaction to get into zkSync

Reply:
Can someone rewrite the paragraph so it is correct please. I am not able to do this.

Reply:
Considering the context, this is what I would write:
"To obtain the cost for sending an L2 to L1 message, please refer [here](https://era.zksync.io/docs/dev/developer-guides/bridging/l1-l2.html#getting-the-base-cost)."

Reply:
@barakshani to be very clear, you are saying, remove this:

"The fee for a transaction is equal to `txBaseCost * gasPrice`. The `gasPrice` is the gas price of the user's transaction, while `txBaseCost` is the base cost for the transaction and depends on the transaction parameters (e.g. `gasLimit` for an `Execute` transaction)."

In favour of this:

"To obtain the cost for sending an L2 to L1 message, please refer [here](https://era.zksync.io/docs/dev/developer-guides/bridging/l1-l2.html#getting-the-base-cost)."

?

Reply:
I do

Reply:
OK, I have just made this change and it now has no relation to the priority queue section at all. It is like a disembodied sentence.

"
## Priority queue

1. All transactions types are now supported by the priority queue. 

2. The priority queue must be fully permissionless to prevent malicious activity. For example, malicious users might send multiple transactions which push up the block gas limit to unworkable levels. To mitigate against this, submitting transactions to the priority queue is no longer free and users must pay a fee to the operator. 

To obtain the cost for sending an L2 to L1 message, please refer to [step 4 of how to send an L1 to L2 transaction](../../how-to/send-transaction-l1-l2.md#step-by-step).
"

On this page, we already have links to these instructions so I'm just going to remove this sentence and we'll have point 1, point 2, and that's it.

Reply:
please check latest commit -> also some of the other please from yesterday

Reply:
ok I see -- in that case we can put the sentence under point 2. It relates to the cost that are mentioned there.

Reply:
ok added back